avogadro/qtgui/meshgenerator.cpp

```diff
@@ -47,66 +47,532 @@ bool MeshGenerator::initialize(const Cube* cube_, Mesh* mesh_, float iso,
+            {
+                size_t idx = ge1.zstart + z1counter;
+                if(isXEnd and isYEnd)
+                {
+                    std::array<float, 3> interpolatedPoint = interpolateOnCube(pointCube, isovalCube, 11);
+
+                    std::array<float, 3> interpolatedNormal = interpolateOnCube(gradCube, isovalCube, 11);                                                      
+
+                    points[idx] = Vector3f(interpolatedPoint[0], interpolatedPoint[1], interpolatedPoint[2]);
+                    normals[idx] = Vector3f(interpolatedNormal[0], interpolatedNormal[1], interpolatedNormal[2]);
+                }
+                globalIdxs[11] = idx;
+            }
+
+            if(isCutCase[5])
+            {
+                size_t idx = ge2.ystart + y2counter;
+                if(isXEnd and isZEnd)
+                {
+                    std::array<float, 3> interpolatedPoint = interpolateOnCube(pointCube, isovalCube, 5);
+
+                    std::array<float, 3> interpolatedNormal = interpolateOnCube(gradCube, isovalCube, 5);                                                      
+
+                    points[idx] = Vector3f(interpolatedPoint[0], interpolatedPoint[1], interpolatedPoint[2]);
+                    normals[idx] = Vector3f(interpolatedNormal[0], interpolatedNormal[1], interpolatedNormal[2]);
+                    // y2 counter does not need to be incremented.
+                }
+                globalIdxs[5] = idx;
+            }
+
+            if(isCutCase[6])
+            {
+                size_t idx = ge3.xstart + x3counter;
+                if(isYEnd and isZEnd)
+                {
+                    std::array<float, 3> interpolatedPoint = interpolateOnCube(pointCube, isovalCube, 6);
+                    std::array<float, 3> interpolatedNormal = interpolateOnCube(gradCube, isovalCube, 6);                                                      
+
+                    points[idx] = Vector3f(interpolatedPoint[0], interpolatedPoint[1], interpolatedPoint[2]);
+                    normals[idx] = Vector3f(interpolatedNormal[0], interpolatedNormal[1], interpolatedNormal[2]);
+                }
+                globalIdxs[6] = idx;
+                ++x3counter;
+            }
+
+            // Add triangles
+            const char* caseTri = caseTriangles[caseId]; // size 16
+            for(int idx = 0; caseTri[idx] != -1; idx += 3)
+            {
+
+
+                size_t globalIdx = globalIdxs[caseTri[idx]];
+                size_t globalIdx1 = globalIdxs[caseTri[idx + 1]];
+                size_t globalIdx2 = globalIdxs[caseTri[idx + 2]];
+
+
+
+                m_vertices.push_back(points[globalIdx]);
+                m_vertices.push_back(points[globalIdx1]);
+                m_vertices.push_back(points[globalIdx2]);
+
+                m_normals.push_back(normals[globalIdx]);
+                m_normals.push_back(normals[globalIdx1]);
+                m_normals.push_back(normals[globalIdx2]);    
+
```

Comment:
Hi @cryos and @ghutchis , does this part looks okay to you guys. I think I have done all things right but my output comes out to be something like. Any suggestions?

![image](https://github.com/user-attachments/assets/c61220fc-5d56-44d0-ad6a-f904670442d1)